src/Functions/array/arrayJaccardIndex.cpp

```diff
@@ -0,0 +1,161 @@
+        }
+        return std::make_shared<DataTypeNumber<ResultType>>();
+    }
+
+    template <bool is_const_left, bool is_const_right>
+    static void vector(const ColumnArray::Offsets & intersect_offsets, const ColumnArray::Offsets & left_offsets, const ColumnArray::Offsets & right_offsets, PaddedPODArray<ResultType> & res)
+    {
+        size_t left_size;
+        size_t right_size;
+        for (size_t i = 0; i < res.size(); ++i)
+        {
+            if constexpr (is_const_left)
+                left_size = left_offsets[0];
+            else
+                left_size = left_offsets[i] - left_offsets[i - 1];
+            if constexpr (is_const_right)
+                right_size = right_offsets[0];
+            else
+                right_size = right_offsets[i] - right_offsets[i - 1];
+
+            size_t intersect_size = intersect_offsets[i] - intersect_offsets[i - 1];
+            res[i] = static_cast<ResultType>(intersect_size) / (left_size + right_size - intersect_size);
+            if (unlikely(isnan(res[i])))
+                res[i] = 1;
+        }
+    }
+
+    template <bool is_const_left, bool is_const_right = false>
+    static void vectorWithEmptyIntersect(const ColumnArray::Offsets & left_offsets, const ColumnArray::Offsets & right_offsets, PaddedPODArray<ResultType> & res)
+    {
+        size_t left_size;
+        size_t right_size;
+        for (size_t i = 0; i < res.size(); ++i)
+        {
+            if constexpr (is_const_left)
+                left_size = left_offsets[0];
+            else
+                left_size = left_offsets[i] - left_offsets[i - 1];
+            if constexpr (is_const_right)
+                right_size = right_offsets[0];
+            else
+                right_size = right_offsets[i] - right_offsets[i - 1];
+
+            res[i] = static_cast<ResultType>(left_size + right_size == 0);
+        }
+    }
+
+    ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr &, size_t input_rows_count) const override
+    {
+        bool is_const_left;
+        bool is_const_right;
+        const ColumnArray * left_array;
+        const ColumnArray * right_array;
+
+        auto cast_array = [&](const ColumnWithTypeAndName & col)
+        {
+            const ColumnArray * res;
+            bool is_const = false;
+            if (typeid_cast<const ColumnConst *>(col.column.get()))
+            {
+                res = checkAndGetColumn<ColumnArray>(checkAndGetColumnConst<ColumnArray>(col.column.get())->getDataColumnPtr().get());
+                is_const = true;
+            }
+            else if (!(res = checkAndGetColumn<ColumnArray>(col.column.get())))
+                throw Exception(ErrorCodes::ILLEGAL_COLUMN, "Argument for function {} must be array but it has type {}.",
+                        col.column->getName(), getName());
+            return std::make_pair(res, is_const);
+        };
+
+        std::tie(left_array, is_const_left) = cast_array(arguments[0]);
```

Comment:
Let's try to use structured bindings instead of `std::tie` ([here](https://www.cppstories.com/2022/structured-bindings/)).